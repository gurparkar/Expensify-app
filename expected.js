import selectExpenses from './src/selectors/expenses'
import moment from 'moment'

const expenses = [{
      id: '1',
      description: 'Gum',
      note: '',
      amount: 195,
      createdAt: moment(0).valueOf()
}, {
    id: '2 ',
    description: 'Rent',
    note: '',
    amount: 109500,
    createdAt: moment(0).add(4, 'days').valueOf()
}, {
    id: '3',
    description: 'Credit Card',
    note: '',
    amount: 4500,
    createdAt: moment(0).add(4, 'days').valueOf()
}]


const filters = {
          text: '',
          sortBy: 'date',
          startDate: moment(-100000),
          endDate: undefined
}
const result = selectExpenses(expenses, filters)
console.log(result)

     
