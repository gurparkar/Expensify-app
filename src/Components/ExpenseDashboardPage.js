import React from 'react'
import  ExpenseList from './ExpenseList'

import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        This is from my dashboard component
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage