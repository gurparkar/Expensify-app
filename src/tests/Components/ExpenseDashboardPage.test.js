import React from 'react'
import {expenses} from '../../../expected'

import { shallow } from 'enzyme'
import  ExpenseDashboardPage  from '../../Components/ExpenseDashboardPage'



test("Testing the ExpenseList Component using Jest and Enzyme", () =>{
    const wrapper =  shallow(<ExpenseDashboardPage />)
    
    expect(wrapper).toMatchSnapshot()

})

