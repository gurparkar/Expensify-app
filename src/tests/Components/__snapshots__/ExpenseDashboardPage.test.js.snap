// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ExpenseList Component using Jest and Enzyme 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  This is from my dashboard component
  <Connect(ExpenseList) />
</div>
`;
